.github/workflows/pr-cleanup.yaml

```diff
@@ -0,0 +1,41 @@
+name: Unpublish Pull Request NPM tag
+
+on:
+  pull_request_target:
+    types:
+      - closed
+
+jobs:
+  unpublish-pr:
+    name: "Unpublish Pull Request NPM tag"
+    runs-on: ubuntu-latest
+    permissions: write-all
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: CI Setup
+        uses: ./.github/actions/ci-setup
+
+      - name: Ensure NPM access
+        run: npm whoami
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Get package versions for PR tag
+        id: pr_versions
+        run: |
+          VERSIONS=$(npm view fuels --json)
+          PR_VERSIONS=$(echo "$VERSIONS" | tr ',' '\n' | tr -d '[]"' | grep -F "pr-${{ github.event.pull_request.number }}" | xargs)
+          echo "name=versions::$PR_VERSIONS" >> $GITHUB_OUTPUT
+      - name: Unpublish NPM PR Tag Versions
+        if: steps.pr_versions.outputs.versions != ''
+        run: |
+          for VERSION in ${{ steps.pr_versions.outputs.versions }}
+          do
+            echo "Unpublishing version $VERSION"
+            npm --force unpublish fuels@${VERSION}
```

Comment:
@Torres-ssf Thank you for all tests and validations that resulted in the suggested version's iteration.

However, this will not be enough because this is limited to the `fuels` package only, whereas _all_ workspaces' packages are getting published. I believe the steps for a complete solution should look something like this:

 1. Iterate over **_all_** workspaces present in `pnpm-workspaces.yaml`
 1. List **_all_** packages in each workspace
 1. Check which ones are being **_published_**
 1. Fetch **_all_** published versions for each one
 1. Finally, **_unpublish_** them all **_individually_**

The complications here tell me that this will be a very risky and challenging task.

In fact, it makes me question how feasible this is and if it's worth the trouble at the moment.

Let's think about this more carefully before we move on.

Reply:
I think we might be able to do something like this :
```
pnpm list -r --depth=-1 | awk '{ if($0 !~ /^[[:space:]]*$/) print $1}'
```
to get the list of names and then we need to iterate through them. but from what I've seen, some of the packages return an error for the npm view, so we need to be attentive of that as well.
(example forc-bin)
@Torres-ssf could you show some docs of how to set up the npm package myself, I'd really want to have a good testing platform before going further with this.

Reply:
@tibi77 Great insight! 🚀 

Not _all_ packages are currently published, these should be marked as private:

https://github.com/FuelLabs/fuels-ts/blob/dadc5760549564480b7b580cb5c268030bf7dcd6/packages/forc-bin/package.json#L2

Reply:
Ahhh makes sense now, ok, will try to hack something together :D

Reply:
I think I got it :)